### PR TITLE
thunderbolt: fix host controller probing

### PR DESCRIPTION
**Pull request by djcampello**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
This seems to break self tests, can you check them?

File: plugins/thunderbolt/fu-thunderbolt-device.c [hunk 1]
```diff
@@ -271,15 +271,22 @@ fu_thunderbolt_device_probe(FuDevice *device, GError **error)
...
 
+	self->devpath = g_strdup(fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device)));
+
 	/* device */
 	if (g_strcmp0(tmp, "thunderbolt_device") == 0) {
+		g_autofree gchar *parent_name =
+		    fu_udev_device_get_parent_name(FU_UDEV_DEVICE(self));
+		/* determine if host controller or not */
+		if (parent_name != NULL && g_str_has_prefix(parent_name, "domain"))
+			self->device_type = FU_THUNDERBOLT_DEVICE_TYPE_HOST_CONTROLLER;
```
Comment (superm1) [inline plugins/thunderbolt/fu-thunderbolt-device.c:282]: doesn't this exact same code already run in `fu_thunderbolt_device_setup_controller`?

Comment (djcampello) [reply plugins/thunderbolt/fu-thunderbolt-device.c:282]: I am moving it on this patch earlier so I can stop failing on device_setup if we cannot read the version

File: plugins/thunderbolt/fu-thunderbolt-device.c [hunk 2]
```diff
@@ -356,10 +356,9 @@ fu_thunderbolt_device_setup_controller(FuDevice *device, GError **error)
 	fu_device_set_name(device, tmp);
 
 	/* set vendor string */
-	tmp = fu_udev_device_get_sysfs_attr(FU_UDEV_DEVICE(self), "vendor_name", error);
-	if (tmp == NULL)
-		return FALSE;
-	fu_device_set_vendor(device, tmp);
+	tmp = fu_udev_device_get_sysfs_attr(FU_UDEV_DEVICE(self), "vendor_name", NULL);
+	if (tmp != NULL)
+		fu_device_set_vendor(device, tmp);
```
Comment (superm1) [inline plugins/thunderbolt/fu-thunderbolt-device.c:361]: Won't this be problematic for targeting firmware if we don't have a vendor ID set?  IIRC that's the reason it was mandatory.

**Comment by superm1** [comment]
> How do you manually run them?

`ninja -C build test`

File: plugins/thunderbolt/fu-thunderbolt-device.c (hunk 2 above)
Comment (djcampello) [reply plugins/thunderbolt/fu-thunderbolt-device.c:361]: What I'm trying to accomplish is to have a host controller show up as a device.. It does not necessarily needs to be an updatable device. This is to enable the use case in https://github.com/fwupd/fwupd/issues/3356 where we would use the host controller to enable the children retimers for probing.

cc @RajmohanMani @krankun

**Comment by djcampello** [comment]
> ninja -C build test

Now I need to figure out this in the context of Chrome OS portage

**Comment by djcampello** [comment]
> This seems to break self tests, can you check them?

Fixed now

File: plugins/thunderbolt/fu-thunderbolt-device.c (hunk 2 above)
Comment (djcampello) [reply plugins/thunderbolt/fu-thunderbolt-device.c:361]: Isn't this just a string for information purposes? The vendor id is set a few lines below: 386/390

Comment (hughsie) [reply plugins/thunderbolt/fu-thunderbolt-device.c:361]: Yup, Vendor is just for the UI. The vendor-id is the important one.

**Review by superm1 — APPROVED** [review]

**Pull request merged**